src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,319 @@
+use crate::{
+    commands::toolchain::ExportCommand,
+    constants::{FUEL_TOOLCHAIN_TOML_FILE, VALID_CHANNEL_STR},
+    path::get_fuel_toolchain_toml,
+    toolchain::{DistToolchainDescription, Toolchain},
+    toolchain_override::{self, OverrideCfg, ToolchainCfg, ToolchainOverride},
+    util::version::exec_version,
+};
+use anyhow::{bail, Result};
+use component::{self, Components};
+use forc_tracing::println_warning;
+use semver::Version;
+use std::{collections::HashMap, path::PathBuf};
+use std::{fs, io::BufRead, str::FromStr};
+use tracing::info;
+
+pub fn export(command: ExportCommand, mut reader: impl BufRead) -> Result<()> {
+    let ExportCommand {
+        name,
+        channel,
+        force,
+    } = command;
+    let mut toolchain_info_path = PathBuf::from("./").join(FUEL_TOOLCHAIN_TOML_FILE);
+    if let Some(path) = get_fuel_toolchain_toml() {
+        toolchain_info_path = path;
+        if force {
+            println_warning(&format!(
+                "Because the `--force` argument was supplied, this file will be overwritten: {}",
+                &toolchain_info_path.display(),
+            ));
+            if fs::remove_file(&toolchain_info_path).is_err() {
+                bail!("Failed to remove file {}", &toolchain_info_path.display());
+            }
+        } else {
+            println_warning(&format!(
+                "There is an existing toolchain override file at {}. \
+                Do you wish to replace it with a new one? (y/N) ",
+                &toolchain_info_path.display(),
+            ));
+            let mut need_replace = String::new();
+            if reader.read_line(&mut need_replace).is_err() {
+                bail!("Failed to read user input");
+            }
+            if need_replace.trim() == "y" {
```

Comment:
> I've recently done a little `stdin` reading. Does this remove spaces and covers new lines? \n and \r?

I think this could be confirmed in the unit test

![image](https://github.com/FuelLabs/fuelup/assets/25278203/c6a20acd-b65a-4189-aabd-c40037447caf)